### ISSUE TITLE
NullPointerException when too much firefoxes are launched

### ISSUE THREAD

**Issue by palladium-dkt** [SeleniumHQ/selenium#4036]
## Meta -
OS:  Centos linux version 2.6.32, tested and reproduced also on Windows 10
Selenium Version:  3.4.0 (java) + geckodriver 0.16.1

Browser:  Firefox

Browser Version:  45.8.0 (linux) 47.0.2 (windows)

## Expected Behavior - 
According to  waitUntilAvailable function in DriverService class, a WebDriverException should be thrown when Firefox driver server doesn't answer within 20 seconds.

```
 protected void waitUntilAvailable() throws MalformedURLException {
    try {
      URL status = new URL(url.toString() + "/status");
      new UrlChecker().waitUntilAvailable(20, SECONDS, status);
    } catch (UrlChecker.TimeoutException e) {
      process.checkForError();
      throw new WebDriverException("Timed out waiting for driver server to start.", e);
    }
  }
```

## Actual Behavior -
When process is not correctly started or stopped within 20sec, "process" variable is null so a NullPointerException is thrown on the process.checkForError() line

> 
> Caused by: java.lang.NullPointerException
> 	at org.openqa.selenium.remote.service.DriverService.waitUntilAvailable(DriverService.java:192)
> 	at org.openqa.selenium.firefox.XpiDriverService.start(XpiDriverService.java:94)
> 	at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:78)
> 	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:637)
> 	... 27 more

## Steps to reproduce -
It happens after launching many firefoxes tests in parallel  (8 on my computer)

**Comment by navels** [comment]
This is causing intermittent failures in our CI environment, but we're only launching one firefox instance.

**Comment by barancev** [comment]
Fixed by c221a3999084d4eed1b51ffe9e43626aca116058, should be released in version 3.5
